components/formats-gpl/src/loci/formats/filter/LZ4Filter.java

```diff
@@ -0,0 +1,75 @@
+package loci.formats.filter;
+
+import ucar.nc2.filter.FilterProvider;
+import ucar.nc2.filter.Filter;
+
+import io.airlift.compress.lz4.Lz4Decompressor;
+
+import java.util.Map;
+import java.io.IOException;
+import java.nio.ByteBuffer;
+
+public class LZ4Filter extends Filter {
+
+  static final String name = "LZ4 filter";
+  static final int id = 32004;
+  private Lz4Decompressor decompressor;
+
+  public LZ4Filter(Map<String, Object> properties) {
+    decompressor = new Lz4Decompressor();
+  }
+
+  @Override
+  public String getName() {
+    return name;
+  }
+
+  @Override
+  public int getId() {
+    return id;
+  }
+
+  @Override
+  public byte[] encode(byte[] dataIn) throws IOException {
+    // only decoding implementation needed for reading
+    byte[] dataOut = dataIn;
```

Comment:
Slight preference for throwing `UnsupportedOperationException` here instead of returning the input array, so that it's very clear this isn't supported.